### PR TITLE
[WFCORE-3731] removal of unused variables aka dead local store

### PR DESCRIPTION
**Pull request by rsvoboda**
Removal of unused variables aka dead local store, using https://issues.jboss.org/browse/WFCORE-3731 as the tracker for these changes.

Focus on main classes, just few tests updated.
Fixing all tests will take some time.

### CODE REVIEW

File: host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncModelHandlerBase.java [hunk 1]
```diff
@@ -86,7 +85,6 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
         final Resource transformedResource = ReadMasterDomainModelUtil.createResourceFromDomainModelOp(localModel.get(RESULT), localExtensions);
 
         // Create the local describe operations
-        final OrderedChildTypesAttachment orderedChildTypesAttachment = new OrderedChildTypesAttachment();
```
Comment (bstansberry) [inline host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncModelHandlerBase.java]: I would prefer that this file and SyncModelOperationHandlerWrapper be left alone. It's very complex stuff and I'd rather we see IDE warns about these next time we have to touch them, as that may spark some investigation as to why they were left behind or lead people to understand the history of the code.

**Comment by rsvoboda** [comment]
>> WFCORE-3731 is already resolved, so we don't expect more PRs against it.

I added info about this PR a changes there to the description and GH PR section, but forgot to reopen it. It won't be used for any other PR and I will create new JIRA for new PRs.

File: core-model-test/framework/src/main/java/org/jboss/as/core/model/test/LegacyKernelServicesImpl.java [hunk 2]
```diff
@@ -137,7 +137,6 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 try {
                     // We get the model as a list of resources descriptions
                     final ModelNode domainModel = operation.get(DOMAIN_MODEL);
-                    final ModelNode startRoot = Resource.Tools.readModel(context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS));
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply core-model-test/framework/src/main/java/org/jboss/as/core/model/test/LegacyKernelServicesImpl.java]: It just reads root resource, didn't anticipate any troubles with it. I will revert this change.

File: host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java [hunk 3]
```diff
@@ -284,7 +284,6 @@ public void handleRequest(DataInput input, ActiveOperation.ResultHandler<Void> r
         @Override
         public void handleRequest(final DataInput input, final ActiveOperation.ResultHandler<Void> resultHandler, final ManagementRequestContext<RegistrationContext> context) throws IOException {
             final byte status = input.readByte();
-            final String message = input.readUTF(); // Perhaps use message when the host failed
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java]: I thought `input.readByte()` line above is sufficient to have.

When you mentioned this I remembered some issue in domain when messages between DC and HC were not fully read and it caused some issues.

Thanks for pointing this out, I will revert the change.

File: host-controller/src/main/java/org/jboss/as/host/controller/mgmt/ServerToHostProtocolHandler.java [hunk 4]
```diff
@@ -335,7 +335,6 @@ private ServerStartedHandler(String serverProcessName) {
         @Override
         public void handleRequest(final DataInput input, final ActiveOperation.ResultHandler<Void> resultHandler, final ManagementRequestContext<ServerInventory> context) throws IOException {
             final byte param = input.readByte(); // Started / Failed PARAM_OK
-            final String message = input.readUTF(); // Server started/failed message
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply host-controller/src/main/java/org/jboss/as/host/controller/mgmt/ServerToHostProtocolHandler.java]: I will revert the change.

File: patching/src/test/java/org/jboss/as/patching/runner/RemoveModifiedFileTaskTestCase.java [hunk 5]
```diff
@@ -110,7 +111,8 @@ public void tearDown() {
     @Test
     public void testRemoveModifiedFileWithSTRICT() throws Exception {
         try {
-            PatchingResult result = runner.applyPatch(zippedPatch, ContentVerificationPolicy.STRICT);
+            runner.applyPatch(zippedPatch, ContentVerificationPolicy.STRICT);
+            fail("Patch shouldn't be applied - ContentVerificationPolicy.STRICT");
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply patching/src/test/java/org/jboss/as/patching/runner/RemoveModifiedFileTaskTestCase.java]: OK, couldn't help myself to fix the test :)

File: host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncModelHandlerBase.java (hunk 1 above)
Comment (rsvoboda) [reply host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncModelHandlerBase.java]: I will revert the change.

**Pull request merged**